tap_pardot/client.py

```diff
@@ -97,8 +83,8 @@ def _make_request(self, method, url, params=None):
             params,
         )
 
-        response = self.requests_session.request(
-            method, url, headers=self._get_auth_header(), params=params
+        response = requests.request(
```

Comment:
I would keep the session object regardless. Always try and re-use a session at the top.